backend/src/main/java/gov/cdc/usds/simplereport/api/devicetestperformedloinccode/DeviceTestPerformedLoincCodeDataResolver.java

```diff
@@ -0,0 +1,36 @@
+package gov.cdc.usds.simplereport.api.devicetestperformedloinccode;
+
+import gov.cdc.usds.simplereport.api.supporteddisease.SupportedDiseaseDataLoaderService;
+import gov.cdc.usds.simplereport.db.model.DeviceTestPerformedLoincCode;
+import gov.cdc.usds.simplereport.db.model.SupportedDisease;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import org.dataloader.DataLoader;
+import org.springframework.graphql.data.method.annotation.SchemaMapping;
+import org.springframework.graphql.execution.BatchLoaderRegistry;
+import org.springframework.stereotype.Controller;
+import reactor.core.publisher.Mono;
+
+@Controller
+public class DeviceTestPerformedLoincCodeDataResolver {
+
+  public DeviceTestPerformedLoincCodeDataResolver(
+      BatchLoaderRegistry registry,
+      SupportedDiseaseDataLoaderService supportedDiseaseDataLoaderService) {
+    registry
+        .forTypePair(UUID.class, SupportedDisease.class)
+        .withName("supportedDiseaseDataLoader")
+        .registerMappedBatchLoader(
+            (supportedDiseaseId, batchLoaderEnvironment) ->
+                Mono.just(
+                    supportedDiseaseDataLoaderService.getSupportedDisease(supportedDiseaseId)));
+  }
+
+  @SchemaMapping(typeName = "SupportedDiseaseTestPerformed", field = "supportedDisease")
+  public CompletableFuture<SupportedDisease> supportedDisease(
+      DeviceTestPerformedLoincCode deviceTestPerformedLoincCode,
+      DataLoader<UUID, SupportedDisease> supportedDiseaseDataLoader) {
+    return supportedDiseaseDataLoader.load(
+        deviceTestPerformedLoincCode.getSupportedDisease().getInternalId());
```

Comment:
This will cause n+1 issue, the moment you perform a deviceTestPerformedLoincCode.getSupportedDisease(), the supported disease will be loaded separately, then you are taking the id of that entity and loading it again.

check this out for reference on how to data load without invoking the getSupportedDisease(), happy to pair on it.

https://github.com/CDCgov/prime-simplereport/blob/c8c3a33d692049b660f374795a358b9b75b9b00a/backend/src/main/java/gov/cdc/usds/simplereport/api/devicetype/DeviceTypeDataResolver.java#L42

Reply:
I am honestly not sure that we need a dataloader for `SupportedDiseaseTestPerformed`, I think the 

I think its better to make the `deviceType` and `supportedDisease` on `DeviceTestPerformedLoincCode` **eagerly** loaded, because there is no value to that object without these two fields, and then adding a dataloader for `DeviceType.supportedDiseaseTestPerformed` as indicated earlier